indexing-service/src/test/java/io/druid/indexing/overlord/TaskLifecycleTest.java

```diff
@@ -273,24 +328,53 @@ public void setUp() throws Exception
     publishCountDown = new CountDownLatch(1);
     announcedSinks = 0;
     pushedSegments = 0;
-    tmp = Files.createTempDir();
+    tmpDir = temporaryFolder.newFolder();
 
     final TaskQueueConfig tqc = new DefaultObjectMapper().readValue(
-        "{\"startDelay\":\"PT0S\", \"restartDelay\":\"PT1S\"}",
+        "{\"startDelay\":\"PT0S\", \"restartDelay\":\"PT1S\", \"storageSyncRate\":\"PT0.5S\"}",
         TaskQueueConfig.class
     );
     indexSpec = new IndexSpec();
-    ts = new HeapMemoryTaskStorage(
-        new TaskStorageConfig(null)
-        {
-        }
-    );
+
+    if (taskStorageType.equals("HeapMemoryTaskStorage")) {
```

Comment:
i think it is better to use parameterized task storage objects instead of string comparisons and then creating the object